### PR TITLE
Make options hash optional on `DS.Model.destroyRecord()`

### PR DESCRIPTION
**Pull request by gossi**
See API docs here: https://emberjs.com/api/ember-data/3.0/classes/DS.Model/methods/destroyRecord?anchor=destroyRecord

Also it's not explictely listet in the API docs to be optional, check the example code below. Also source code for this: https://github.com/emberjs/data/blob/master/addon/-private/system/model/model.js#L618

Which passes `options` to the `save()` method, which is declared here: https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/ember-data/index.d.ts#L513

### CODE REVIEW

**Review by dwickern — APPROVED** [review]

**Comment by andy-ms** [comment]
Approved by a listed owner. PR ready to merge pending express review by a maintainer.

**Pull request merged**
